documentation/Extension_Point_WebScripts.md

```diff
@@ -0,0 +1,306 @@
+
+This is a simple interface, capable of handling a webrequest by itself:
+
+```java
+public interface Resolution {
+    void resolve(final AnnotationWebScriptRequest request, final AnnotationWebscriptResponse response,
+                 final ResolutionParameters params) throws Exception;
+}
+```
+
+Whenever your return a model from your Uri method, you're implicitly returning a `TemplateResolution` from your method.
+
+The power of a Resoluton comes with reuse:
+if your webscript needs to output JSON, you can return an anonymous implementation of `JsonWriterResolution`:
+
+
+```java
+@Uri("/api/hello")
+public Resolution sayHello() {
+  return new JsonWriterResolution() {
+    void writeJson(JSONWriter jsonWriter) throws JSONException {
+      jsonWriter.object()
+        .key("message").value("hello")
+        .endObject();
+    }
+  };
+}
+```
+
+If you want to use another Json serialisation library, create a new subclass of `JsonResolution`.
+
+Because resolutions are returned, they automatically control the flow of your Uri method.
+
+## Request and ResponseBody
+
+Since version 1.7, there is support for the Spring [RequestBody](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/RequestBody.html) and [ResponseBody](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/ResponseBody.html) annotation. These work with using HttpMessageConverters. At the moment, there are only 2 HttpMessageConverterts available by default. The Jackson converter for json, and the JAXB converter for xml.
+
+### RequestBody
+The `@RequestBody` annotation will _resolve the method argument depending on the content type of the request._ When annotating a method parameter with this, the body of the request will be automatically resolved to the parameter object. Dynamic extensions will select a resolver based on the `Content-Type` header from the request. If no resolver for the specified content-type can be found, a `RuntimeException` will be thrown. 
+
+```java
+@Uri(value = "/example", method = HttpMethod.POST)
+public void exampleRequestBody(@RequestBody MyObject thing) {
+    // do some work
+}
+```
+
+There is also support for the `required` parameter in the annotation. When `required = true`, an exception will be thrown of the body of the request is empty.
+
+### ResonseBody
+When annotating a webscript with the `@ResponseBody` annotation, the _method return value should be bound to the web response body_. This means that when your webscript returns an Object, that object should be serialized to a data format that the client accepts. This format has to be specified in the `Accept` header from the request. If no accept header is present, the webscript default format will be used.
+
+```java
+@Uri(value = "/getPerson", defaultFormat = MediaType.APPLICATION_JSON_VALUE)
+@ResponseBody
```

Comment:
Is this correct? I thought the `@ResponseBody` needed to annotate the specific type which it serializes. Like so:
```java
@Uri(value = "/getPerson", defaultFormat = MediaType.APPLICATION_JSON_VALUE)
public @ResponseBody Person handleDefaultResponse(…
```

Reply:
This is exactly the same. You do not have annotations on return types, they are always annotations on the method.

Reply:
I wasn't aware, thanks!

Reply:
This is exactly the same, `@ResponseBody` is a method annotation. Also methods can only have a single return type?

Reply:
Yeah it's fine I was just confused on how annotations work in Java.